modules/objdetect/src/face_recognize.cpp

```diff
@@ -0,0 +1,186 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+
+#include "opencv2/imgproc.hpp"
+#include "opencv2/core.hpp"
+#include "opencv2/dnn.hpp"
+
+#include <algorithm>
+
+namespace cv
+{
+
+class DNNFaceRecognizerImpl : public DNNFaceRecognizer
+{
+public:
+    DNNFaceRecognizerImpl(const String& onnx_path)
+    {
+        this->model = dnn::readNet(onnx_path);
+    };
+    void AlignCrop(InputArray _src_img, InputArray _face_mat, OutputArray _aligned_img) const override
+    {
+        Mat face_mat = _face_mat.getMat();
+        float src_point[5][2];
+        for (int row = 0; row < 5; ++row)
+        {
+            for(int col = 0; col < 2; ++col)
+            {
+                src_point[row][col] = face_mat.at<double>(1, row*2+col+4);
```

Comment:
float, not double (reason for all the NaN's)

Reply:
Done at https://github.com/fengyuentau/opencv/commit/8779ccb3c0ee9bc9c5cb522150bb8cbef6801d92.

Reply:
Done at https://github.com/fengyuentau/opencv/commit/8779ccb3c0ee9bc9c5cb522150bb8cbef6801d92.